src/k8s/pkg/k8sd/setup/certificates_test.go

```diff
@@ -126,9 +121,79 @@ func TestEnsureWorkerPKI(t *testing.T) {
 	}
 
 	for _, file := range expectedFiles {
-		_, err := os.Stat(file)
-		if err != nil {
+		if _, err := os.Stat(file); err != nil {
 			t.Errorf("Expected file %q is missing: %v", file, err)
 		}
 	}
 }
+
+func TestExtDatastorePKI(t *testing.T) {
+	tempDir := t.TempDir()
+	mock := &mock.Snap{
+		Mock: mock.Mock{
+			EtcdPKIDir: tempDir,
+			UID:        os.Getuid(),
+			GID:        os.Getgid(),
+		},
+	}
+	certificates := &pki.ExternalDatastorePKI{
+		DatastoreCACert:     "ca_cert",
+		DatastoreClientKey:  "client_key",
+		DatastoreClientCert: "client_cert",
+	}
+
+	if _, err := setup.EnsureExtDatastorePKI(mock, certificates); err != nil {
+		t.Fatalf("EnsureExtDatastorePKI returned unexpected error: %v", err)
+	}
```

Comment:
extend these tests, to test that the `changed` value is what we expect. this would have already caught some of the comments above.

Reply:
Consider creating a `certificates_internal_test.go` with a `Test_ensureFiles()` to add extensive coverage